src/Illuminate/Support/Number.php

```diff
@@ -342,6 +343,26 @@ public static function trim(int|float $number)
         return json_decode(json_encode($number));
     }
 
+    public static function roman(int $number): string
+    {
+        if ($number < 1 || $number > 3999) {
+            throw new InvalidArgumentException('Number must be between 1 and 3999');
+        }
+
+        $values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
+        $symbols = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I'];
+
+        $roman = '';
+        for ($i = 0; $i < count($values); $i++) {
+            while ($number >= $values[$i]) {
+                $number -= $values[$i];
+                $roman .= $symbols[$i];
+            }
+        }
```

Comment:
With two separate array for `$values` and `$symbols`, there's no need for a `for` loop:

```suggestion
        $values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
        $symbols = ['M', 'CM', 'D', 'CD', 'C', 'XC', 'L', 'XL', 'X', 'IX', 'V', 'IV', 'I'];

        $roman = '';
        foreach ($values as $i => $value) {
            while ($number >= $value) {
                $number -= $value;
                $roman .= $symbols[$i];
            }
        }
```
If you don't want a `foreach` loop instead, there's no need for two separate arrays:

```suggestion
        $mapping = [
            1000 => 'M',
            900 => 'CM',
            500 => 'D',
            400 => 'CD',
            100 => 'C',
            90 => 'XC',
            50 => 'L',
            40 => 'XL',
            10 => 'X',
            9 => 'IX',
            5 => 'V',
            4 => 'IV',
            1 => 'I',
        ];

        $roman = '';
        while ($symbol = current($mapping)) {
            $value = key($mapping);
            while ($number >= $values[$i]) {
                $number -= $value;
                $roman .= $symbol;
            }
        }
```

Reply:
> With two separate array for `$values` and `$symbols`, there's no need for a `for` loop:
> 
> If you don't want a `foreach` loop instead, there's no need for two separate arrays:

You're right, it's just my habit. But using `foreach` looks better